streams/src/test/java/org/apache/kafka/streams/processor/UsePreviousTimeOnInvalidTimestampTest.java

```diff
@@ -42,4 +43,12 @@ public void usePreviousTimeOnInvalidTimestamp() {
         assertThat(timestamp, is(previousTime));
     }
 
+    @Test(expected = StreamsException.class)
```

Comment:
It would be better to only wrap the call to `.extract` -- otherwise, it's unclear where the exception is thrown.

```
public void shouldThrowStreamsException() {
    final TimestampExtractor extractor = new UsePreviousTimeOnInvalidTimestamp();
    final ConsumerRecord record = new ConsumerRecord<>("anyTopic", 0, 0, null, null);
    try {
        extractor.extract(record, -1);
        fail("should have thrown StreamsException");
    } catch (final StreamException expected) { }
}
```